### ISSUE TITLE
Error reading archive

### ISSUE THREAD

**Issue by sidred** [Majored/rs-async-zip#64]
Unable to read an archive using the code 

```
    let mut file = tokio::fs::File::open(zip_file).await.unwrap();
    let mut zip = ZipFileReader::new(&mut file).await.unwrap();

    let entry = zip.file().entries().get(0).unwrap().clone();
    let mut string = String::new();
    let mut reader = zip.entry(0).await.unwrap();
    let txt = reader.read_to_string_checked(&mut string, entry.entry()).await.unwrap();
    println!("{}", txt);

```
Error is
```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: UpstreamReadError(Custom { kind: InvalidData, error: "stream did not contain valid UTF-8" })'

```

This happens on a simple zip file with two files (a.txt and b.txt) having content
```
01234567890123456789012345678901234567890123456789
```

Extracting the zip file without crc check shows the content of a.txt as
```
45678901234567890123456789
PK
�����XC0V4^3���3�
```

Looks like a regression in v0.0.10. Works correctly in v0.0.9

**Comment by sidred** [comment]
Yeah seems to work with current master. Let me see if i can come up with a quick test case for this.

**Comment by Majored** [comment]
@sidred I've released v0.0.11 if that's any help to you. 😄
